src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp

```diff
@@ -230,7 +265,22 @@ NFTokenAcceptOffer::pay(
     if (amount < beast::zero)
         return tecINTERNAL;
 
-    return accountSend(view(), from, to, amount, j_);
+    auto const result = accountSend(view(), from, to, amount, j_);
+
+    // After this amendment, if any payment would cause a non-IOU-issuer to
+    // have a negative balance, or an IOU-issuer to have a positive balance in
+    // their own currency, we know that something went wrong. This was
+    // originally found in the context of IOU transfer fees. Since there are
+    // several payouts in this tx, just confirm that the end state is OK.
+    if (!view().rules().enabled(fixUnburnableNFToken))
+        return result;
+    if (result != tesSUCCESS)
+        return result;
+    if (accountFunds(view(), from, amount, fhZERO_IF_FROZEN, j_).signum() < 0)
+        return tecINSUFFICIENT_FUNDS;
+    if (accountFunds(view(), to, amount, fhZERO_IF_FROZEN, j_).signum() < 0)
```

Comment:
this is what actually solves issue 2. we aren't checking whether they have enough in addition to IOU transfer fees. so as the safest option, rather than checking it there, we attempt to do the sale but fail if it resulted in negative balances

Reply:
cool. i have few more questions
1. what does the `signum()` here do?
2. it says when something goes wrong, the IOU-issuer has a positive of its own currency. Then what is the "normal" balance for the IOU issuer when everything goes right? Is it just 0? (just a general question regarding to how IOU works)